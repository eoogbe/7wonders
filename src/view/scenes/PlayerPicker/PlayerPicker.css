/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.PlayerPicker {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.PlayerPicker-title {
  font-size: 1.5rem;
  margin: 0;
  margin-bottom: 1em;
}

.PlayerPicker-picker {
  font-size: 1rem;
  display: grid;
  grid: repeat(2, 4em) / repeat(3, 6em);
}

.PlayerPicker-arrow {
  font-size: 3rem;
  padding: 0;
  border: none;
  outline: none;
  cursor: pointer;
}

.PlayerPicker-playerCount {
  font-size: 2.5rem;
  text-align: center;
}

.PlayerPicker-nextButton {
  font-size: 1.5rem;
  grid-column: 2;
}
