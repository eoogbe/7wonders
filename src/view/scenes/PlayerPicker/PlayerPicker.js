/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { compose } from 'ramda';
import { MdKeyboardArrowLeft, MdKeyboardArrowRight } from 'react-icons/md';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from 'model/actions';
import { getPlayerPicker } from 'presentation/playerPicker';
import * as theme from 'view/theme';
import './PlayerPicker.css';

const getArrowStyle = (isEnabled) => ({
  color: isEnabled ? theme.primaryColor: theme.disabledColor,
});

const PlayerPicker = () => {
  const { playerCount, leftArrowEnabled, rightArrowEnabled } =
    useSelector(getPlayerPicker);

  const dispatch = useDispatch();
  const decrementPlayerCount = compose(dispatch, actions.decrementPlayerPicker);
  const incrementPlayerCount = compose(dispatch, actions.incrementPlayerPicker);
  const pickPlayerCount = compose(dispatch, actions.pickPlayerCount);

  return (
    <section className="PlayerPicker">
      <h1 className="PlayerPicker-title">
        Choose the number of players:
      </h1>
      <div className="PlayerPicker-picker">
        <button
          className="PlayerPicker-arrow"
          style={getArrowStyle(leftArrowEnabled)}
          type="button"
          disabled={!leftArrowEnabled}
          onClick={decrementPlayerCount}
        >
          <MdKeyboardArrowLeft />
        </button>
        <div className="PlayerPicker-playerCount">{playerCount}</div>
        <button
          className="PlayerPicker-arrow"
          style={getArrowStyle(rightArrowEnabled)}
          type="button"
          disabled={!rightArrowEnabled}
          onClick={incrementPlayerCount}
        >
          <MdKeyboardArrowRight />
        </button>
        <button
          className="PlayerPicker-nextButton primary-button"
          type="button"
          onClick={pickPlayerCount}
        >
          Next
        </button>
      </div>
    </section>
  );
};

export default PlayerPicker;
