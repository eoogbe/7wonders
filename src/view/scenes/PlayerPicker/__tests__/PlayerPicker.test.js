/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {
  decrementPlayerPicker, incrementPlayerPicker, pickPlayerCount,
} from 'model/actions';
import PlayerPicker from '../PlayerPicker';

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore(createState({ playerCount: 4 }));
  const tree =
    renderer.create(
      <Provider store={store}>
        <PlayerPicker />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('when min count renders a disabled left arrow', () => {
  const store = mockStore(createState({ playerCount: 3 }));
  const tree =
    renderer.create(
      <Provider store={store}>
        <PlayerPicker />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('when max count renders a disabled right arrow', () => {
  const store = mockStore(createState({ playerCount: 7 }));
  const tree =
    renderer.create(
      <Provider store={store}>
        <PlayerPicker />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('decrements the player count', () => {
  const store = mockStore(createState({ playerCount: 4 }));
  const tree = mount(
    <Provider store={store}>
      <PlayerPicker />
    </Provider>
  );
  tree.find('.PlayerPicker-arrow').first().simulate('click');
  expect(store.getActions()).toEqual([decrementPlayerPicker()]);
});

it('increments the player count', () => {
  const store = mockStore(createState({ playerCount: 4 }));
  const tree = mount(
    <Provider store={store}>
      <PlayerPicker />
    </Provider>
  );
  tree.find('.PlayerPicker-arrow').last().simulate('click');
  expect(store.getActions()).toEqual([incrementPlayerPicker()]);
});

it('picks the player count', () => {
  const store = mockStore(createState({ playerCount: 4 }));
  const tree = mount(
    <Provider store={store}>
      <PlayerPicker />
    </Provider>
  );
  tree.find('.PlayerPicker-nextButton').simulate('click');
  expect(store.getActions()).toEqual([pickPlayerCount()]);
});

const createState = ({ playerCount }) => ({
  app: { playerPickerCount: playerCount },
});
