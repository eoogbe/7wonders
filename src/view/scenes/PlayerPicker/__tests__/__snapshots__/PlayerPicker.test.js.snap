// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<section
  className="PlayerPicker"
>
  <h1
    className="PlayerPicker-title"
  >
    Choose the number of players:
  </h1>
  <div
    className="PlayerPicker-picker"
  >
    <button
      className="PlayerPicker-arrow"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
        />
      </svg>
    </button>
    <div
      className="PlayerPicker-playerCount"
    >
      4
    </div>
    <button
      className="PlayerPicker-arrow"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
        />
      </svg>
    </button>
    <button
      className="PlayerPicker-nextButton primary-button"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</section>
`;

exports[`when max count renders a disabled right arrow 1`] = `
<section
  className="PlayerPicker"
>
  <h1
    className="PlayerPicker-title"
  >
    Choose the number of players:
  </h1>
  <div
    className="PlayerPicker-picker"
  >
    <button
      className="PlayerPicker-arrow"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
        />
      </svg>
    </button>
    <div
      className="PlayerPicker-playerCount"
    >
      7
    </div>
    <button
      className="PlayerPicker-arrow"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55, 0.5)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
        />
      </svg>
    </button>
    <button
      className="PlayerPicker-nextButton primary-button"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</section>
`;

exports[`when min count renders a disabled left arrow 1`] = `
<section
  className="PlayerPicker"
>
  <h1
    className="PlayerPicker-title"
  >
    Choose the number of players:
  </h1>
  <div
    className="PlayerPicker-picker"
  >
    <button
      className="PlayerPicker-arrow"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55, 0.5)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
        />
      </svg>
    </button>
    <div
      className="PlayerPicker-playerCount"
    >
      3
    </div>
    <button
      className="PlayerPicker-arrow"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(93, 64, 55)",
        }
      }
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
        />
      </svg>
    </button>
    <button
      className="PlayerPicker-nextButton primary-button"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</section>
`;
